### PR TITLE
Typo fixes and grammar improvements

### PR DESCRIPTION
**Pull request by kevgo**
Just some intentionally lightweight fixes, in particular:

- use active instead of passive language
- link to interfaces mentioned

Please guide me on how you prefer to receive such contributions. I'm happy to make any adjustments, revert particular changes, or break up this PR into several ones. Thanks!

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -28,41 +28,46 @@ Tips on upgrading can be found in [UPGRADE.md](UPGRADE.md)
...
 ### JSON
 
-Herodot supplies an interface, so it's possible to write many outputs, such as XML and others. For now, JSON is supported.
+Herodot supplies an interface, allowing to return errors in many data formats like XML and others. Currently, it supports only JSON.
 
 #### Write responses
 
 ```go
-var writer = herodot.NewJSONWriter(nil)
+var hd = herodot.NewJSONWriter(nil)
```
Comment (kevgo) [inline README.md]: `writer.Write` stutters, so I'm looking for a better name for this variable. It doesn't have to be `hd`.

File: README.md [hunk 2]
```diff
@@ -112,4 +116,4 @@ available. The output of such an error looks like this:
 ```
 
 To add context to your errors, implement `herodot.ErrorContextCarrier`. If you only want to set the status code of errors
```
Comment (kevgo) [inline README.md]: I didn't find `ErrorContextCarrier` anywhere. Maybe it no longer exists?

File: README.md [hunk 3]
```diff
@@ -28,41 +28,46 @@ Tips on upgrading can be found in [UPGRADE.md](UPGRADE.md)
...
 
 #### Write responses
 
 ```go
-var writer = herodot.NewJSONWriter(nil)
+var hd = herodot.NewJSONWriter(nil)
 
 func GetHandler(rw http.ResponseWriter, r *http.Request) {
-	writer.Write(rw, r, map[string]interface{}{
+	// run your business logic here
```
Comment (kevgo) [inline README.md]: I was struggling to understand these examples until later below in the error handling section I realized that these are supposed to be HTTP handlers and typically one doesn't just return a response but calls some business logic here. To make this easier, I'm adding these comments.

File: README.md (hunk 1 above)
Comment (aeneasr) [reply README.md]: I think `hd` makes sense!

File: README.md (hunk 3 above)
Comment (aeneasr) [reply README.md]: Ah yeah no, it doesn't - it was broken up into several private interfaces: https://github.com/ory/herodot/blob/master/error.go#L23-L56

**Comment by aeneasr** [comment]
Sorry, completely forgot to merge this!

**Pull request merged**